resources/js/app.js

```diff
@@ -24,12 +24,28 @@ Vue.mixin(route);
 
 const app = document.getElementById('app');
 
+// Import frequently used (base) components globally & automatically
+let requireComponent = require.context('./components', true, /Base[A-Z]\w+\.(vue|js)$/);
+requireComponent.keys().forEach(function (fileName) {
+    let baseComponentConfig = requireComponent(fileName);
+    let baseComponentName = baseComponentConfig.name || (
+        fileName
+            .split('/')
+            .pop()
+            .replace(/\.\w+$/, '')
+    );
+    Vue.component(
+        baseComponentName,
+        baseComponentConfig.default || baseComponentConfig
+    )
+});
```

Comment:
```suggestion
const baseComponents = require.context('./components', true, /Base[A-Z]\w+\.(vue|js)$/);
baseComponents.keys().forEach(fileName => {
    const baseComponent = baseComponents(fileName);
    const baseComponentName = baseComponent.name || (
        fileName
            .split('/')
            .pop()
            .replace(/\.\w+$/, '')
    );
    Vue.component(
        baseComponentName,
        baseComponent.default || baseComponent
    )
});
```

I think this is a bit more beautiful ;-)